subiquity/tests/api/test_api.py

```diff
@@ -2379,3 +2396,204 @@ async def test_post_source_update_server_variant(self):
+                        "wipe": "superblock",
+                        "format": "swap",
+                        "mount": "",
+                        "effectively_encrypted": False,
+                    },
+                    {
+                        "number": 4,
+                        "preserve": False,
+                        "wipe": None,
+                        "effective_format": "zfs",
+                        "effective_mount": "/",
+                        "effectively_encrypted": False,
+                    },
+                ),
+            ),
+            (
+                "ZFS_LUKS_KEYSTORE",
+                "passw0rd",
+                (
+                    {
+                        "number": 1,
+                        "boot": True,
+                        "grub_device": True,
+                        "preserve": False,
+                        "wipe": "superblock",
+                        "format": "fat32",
+                        "mount": "/boot/efi",
+                        "effectively_encrypted": False,
+                    },
+                    {
+                        "number": 2,
+                        "boot": False,
+                        "grub_device": None,
+                        "preserve": False,
+                        "wipe": None,
+                        "effective_format": "zfs",
+                        "effective_mount": "/boot",
+                        "effectively_encrypted": False,
+                    },
+                    {
+                        "number": 3,
+                        "preserve": False,
+                        "wipe": None,
+                        "effective_format": "swap",
+                        "effective_mount": "",
+                        "effectively_encrypted": True,
+                    },
+                    {
+                        "number": 4,
+                        "preserve": False,
+                        "wipe": None,
+                        "effective_format": "zfs",
+                        "effective_mount": "/",
+                        "effectively_encrypted": True,
+                    },
+                ),
+            ),
+        )
+    )
+    @timeout()
+    async def test_labels(self, capability, password, partitions):
+        async with start_server("examples/machines/simple.json") as inst:
+            resp = await inst.get("/storage/v2/guided?wait=true")
+            [reformat, manual] = resp["targets"]
+
+            await inst.post(
+                "/storage/v2/guided",
+                {
+                    "target": reformat,
+                    "capability": capability,
+                    "password": password,
+                },
+            )
+            resp = await inst.get("/storage/v2")
+            [d] = resp["disks"]
+            for expected, actual in zip(partitions, d["partitions"]):
+                self.assertDictSubset(expected, actual, f"partnum {actual['number']}")
```

Comment:
suggestion: We could use `itertools.zip_longest` to guarantee we don't truncate one of the lists. I think the scenario in which we do that is unlikely but I generally favor zip_longest when comparing objects. 

```suggestion
            for expected, actual in itertools.zip_longest(partitions, d["partitions"], fillvalue={}):
                self.assertDictSubset(expected, actual, f"partnum {actual['number']}")
```

Reply:
ah, nice.  That's better, thanks.  I'm going with fillvalue=None as I want to see that case fail - we shouldn't randomly be getting a different number of partitions here and I want that to make itself evident in the failed test.

Reply:
Oh yeah even better. I can see the empty dict allowing some annoying cases!